sdk/storage/Azure.Storage.Blobs/src/Models/GetBlobs.cs

```diff
@@ -0,0 +1,288 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for
+// license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Storage.Blobs.Models
+{
+    /// <summary>
+    /// Specifies options for listing blobs with the 
+    /// <see cref="BlobContainerClient.GetBlobsAsync"/> and
+    /// <see cref="BlobContainerClient.GetBlobsByHierarchyAsync"/>
+    /// operations.
+    /// </summary>
+    public struct GetBlobsOptions : IEquatable<GetBlobsOptions>
+    {
+        /// <summary>
+        /// Gets or sets a string that filters the results to return only
+        /// blobs whose name begins with the specified prefix.
+        /// </summary>
+        public string Prefix { get; set; }  // No Prefix header is produced if ""
+
+        /// <summary>
+        /// Gets or sets the maximum number of blobs to return. If the
+        /// request does not specify <see cref="PageSizeHint"/>, or specifies a
+        /// value greater than 5000, the server will return up to 5000 items.
+        /// 
+        /// Note that if the listing operation crosses a partition boundary,
+        /// then the service will return a <see cref="BlobsFlatSegment.NextMarker"/>
+        /// or <see cref="BlobsHierarchySegment.NextMarker"/> for retrieving
+        /// the remainder of the results.  For this reason, it is possible that
+        /// the service will return fewer results than specified by
+        /// <see cref="PageSizeHint"/>, or than the default of 5000. 
+        /// 
+        /// If the parameter is set to a value less than or equal to zero, 
+        /// a <see cref="StorageRequestFailedException"/> will be thrown.
+        /// </summary>
+        public int? PageSizeHint { get; set; }
+
+        /// <summary>
+        /// Gets or sets a flag specifing that metadata related to any current
+        /// or previous <see cref="Specialized.BlobClient.StartCopyFromUriAsync"/>
+        /// operation should be included.
+        /// </summary>
+        public bool IncludeCopyMetadata { get; set; }
```

Comment:
Should all these booleans just be a flags enum property? Just like in ConfigurationcLient APIs.

Reply:
I think that's reasonable and I'll change it if we add a guideline.